samples: blink_led/fade_led: Update when no on-board PWM LED

erwango opened the pull request:
Propose an alternative solution for to use with blank_led and
fade_led samples when no on-board PWM connected LED is available
on the targeted board.
User will have to provide an overlay file matching his
own set up using a LED connected to a PWM pin on his board.
An example of overlay, tested onnucleo_f401re is provided.

This solves the error reported by user that can't use PWM basic
sample due to board limitations.

Fixes #22091

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

mbolivar-nordic commented:
@erwango do you prefer this to https://github.com/zephyrproject-rtos/zephyr/pull/24949?

erwango commented:
@mbolivar-nordic It's an old PR that I though would be orthogonal to on going DT works.
If you agree on the main idea (propose an easy way for users to test their own board fixture), I can put this on top of #24949.

mbolivar-nordic commented on samples/basic/fade_led/README.rst:
```diff
@@ -55,3 +25,20 @@ for the nrf52840dk_nrf52840 board:
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/blink_led
+   :board: nucleo_f401re
+   :gen-args: -DDTC_OVERLAY_FILE="nucleo_f401re.overlay"
```
This is not necessary. The build system picks up boards/BOARD.overlay automagically.

erwango committed:
samples: blinky_pwm/fade_led: Update when no on-board PWM LED

Propose an alternative solution for to use with blinky_pwm and
fade_led samples when no on-board PWM connected LED is available
on the targeted board.
User will have to provide an overlay file matching his
own set up using a LED connected to a PWM pin on his board.
An example of overlay, tested onnucleo_f401re is provided.

This solves the error reported by user that can't use PWM basic
sample due to board limitations.

Fixes #22091

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@mbolivar rebased

mbolivar-nordic commented on samples/basic/blinky_pwm/README.rst:
```diff
@@ -74,3 +56,19 @@ Change ``nrf52840dk_nrf52840`` appropriately for other supported boards.
+If there is no on-board LED connected to a PWM configured PIN on your board,
+you can wire one manually and use a .overlay file to configure LED-PWM driver.
+An example of .overlay file is available in current folder and works with
+nucleo_f401re.
```
```suggestion
nucleo_f401re. See :ref:`set-devicetree-overlays` for more information.
```

mbolivar-nordic commented on samples/basic/blinky_pwm/README.rst:
```diff
@@ -74,3 +56,19 @@ Change ``nrf52840dk_nrf52840`` appropriately for other supported boards.
+*************************************************************************
+
+If there is no on-board LED connected to a PWM configured PIN on your board,
+you can wire one manually and use a .overlay file to configure LED-PWM driver.
```
```suggestion
you can wire one manually and use a devicetree overlay to configure this application.
```

mbolivar-nordic commented on samples/basic/fade_led/README.rst:
```diff
@@ -18,6 +18,14 @@ Requirements and Wiring
+*************************************************************************
+
+If there is no on-board LED connected to a PWM configured PIN on your board,
+you can wire one manually and use a .overlay file to configure LED-PWM driver.
```
Similar suggestions here.

mbolivar-nordic commented on samples/basic/blinky_pwm/README.rst:
```diff
@@ -74,3 +56,19 @@ Change ``nrf52840dk_nrf52840`` appropriately for other supported boards.
+Alternative configuration for board without on-board LED connected to PWM
+*************************************************************************
+
+If there is no on-board LED connected to a PWM configured PIN on your board,
```
Maybe just put this where the table was?

mbolivar-nordic reviewed (approved):
Looks good, some ideas for documentation suggested.

erwango commented:
@mbolivar-nordic, txs for the review. I've taken your remarks and went a bit further.

mbolivar-nordic commented on samples/basic/blinky_pwm/README.rst:
```diff
@@ -35,29 +35,11 @@ board:
-     - connect PWM2 (PA0) to an LED
-   * - :ref:`nucleo_f103rb_board`
-     - connect PWM1 (PA8) to an LED
+In this case, you can still make this application working on your board by
```
```suggestion
In this case, you can still make this application work on your board by
```

mbolivar-nordic commented on samples/basic/blinky_pwm/README.rst:
```diff
@@ -35,29 +35,11 @@ board:
-   * - :ref:`nucleo_f103rb_board`
-     - connect PWM1 (PA8) to an LED
+In this case, you can still make this application working on your board by
+wiring one manually to an PWM enabled pin and use a devicetree overlay to
```
```suggestion
wiring an LED manually to a PWM enabled pin and using a devicetree overlay to
```

mbolivar-nordic commented on samples/basic/blinky_pwm/README.rst:
```diff
@@ -35,29 +35,11 @@ board:
+In this case, you can still make this application working on your board by
+wiring one manually to an PWM enabled pin and use a devicetree overlay to
+reflect this configuration.
+An example of .overlay file is available in current folder and works with
```
```suggestion
An example .overlay file is available in the boards/ folder and works with
```

mbolivar-nordic reviewed (commented):
Last round of nitpicks, I swear. Just some grammar things.

erwango commented:
@mbolivar-nordic fixed, txs for the review!

anangl commented on samples/basic/blinky_pwm/README.rst:
```diff
@@ -35,29 +35,11 @@ board:
+In this case, you can still make this application work on your board by
+wiring an LED manually to a PWM enabled pin and using a devicetree overlay to
+reflect this configuration.
+An example of .overlay file is available in the /boards folder and works with
```
I guess it was supposed to be "boards/".

erwango replied:
Fixed, txs!

mbolivar-nordic commented:
@erwango did you really mean to close this?

erwango commented:
@mbolivar-nordic, no it wasn't intentional, I was cleaning up my repo branches. Thanks for heads up!

MaureenHelm commented:
Will you please rebase?

erwango commented:
@MaureenHelm done, txs

erwango commented:
Set to DNM as #26035 sets up a pwm-led on board.
I need to find a board on which it can't be done at all to avoid this kind of conflicts

galak commented:
@erwango is this still DNM?

erwango commented:
@galak yes as PWM-LED has been added to this board that was used as example in between.. so I need to change the target and push it again
